### PR TITLE
Telemetry call routing to sunbird

### PR DESCRIPTION
**Pull request by manzarul**

### CODE REVIEW

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/LMAXWriter.java [hunk 1]
```diff
@@ -54,14 +55,15 @@ private void init() {
     if (power % 1 != 0) {
       power = Math.ceil(power);
       ringBufferSize = (int) Math.pow(2, power);
-      ProjectLogger.log("New ring buffer size = " + ringBufferSize);
+      ProjectLogger.log(
+          "LMAXWriter:init new ring buffer size = " + ringBufferSize, LoggerEnum.INFO.name());
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/LMAXWriter.java]: Change ...

From:
"LMAXWriter:init new ring buffer size = "

To:
"LMAXWriter:init: New ring buffer size = "

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package org.sunbird.util.lmaxdisruptor;
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Delete EkstepEventConsumer.java as it is not used.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Why two telemetry pom projects are required in lmw-mw? Remove the one unnecessary.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+import com.google.common.net.HttpHeaders;
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Delete CassandraEventConsumer.java as it is not used.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * This class will send telemetry data to sunbird telemetry service. sunbird telemetry service has
+ * the logic for data storage in sunbird and passing same data to Ekstep telemetry.
+ *
+ * @author Manzarul
+ */
+public class SunbirdTelemetryEventConsumer implements EventHandler<Request> {
+
+  @Override
+  public void onEvent(Request request, long sequence, boolean endOfBatch) throws Exception {
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Throws Exception seems not needed.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  /**
+   * This method will transform incoming requested data to ekstep telemetry data structure.
+   *
+   * @param request this will contains telemetry data generated by Sunbird.
+   * @return Structured telemetry data accepted by Ekstep.
+   */
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  private TelemetryV3Request getEkstepTelemetryRequest(Request request) {
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Rename getEkstepTelemetryRequest to getTelemetryRequest.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        ProjectLogger.log(
+            "SunbirdTelemetryEventConsumer:onEvent exception occured during telemetry data processing.Data = "
+                + new Gson().toJson(getEkstepTelemetryRequest(request)),
+            LoggerEnum.INFO.name());
+      }
+    }
+  }
+
+  private Map<String, String> getHeaders() {
+    Map<String, String> headers = new HashMap<String, String>();
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Change HashMap<String, String> to HashMap<>.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    headers.put(HttpHeaders.CONTENT_TYPE, "application/json");
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: Use suitable constant for "application/json".

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java (hunk 3 above)
Comment (manzarul) [reply actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: i can't see two branch.

File: actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  /**
+   * This method will return telemetry url. it will read sunbird_lms_base_url key for base url .
+   * First it will try to read value from environment in case of absence it will read value from
+   * property cache. It will read the uri from sunbird_lms_telemetry_api_url key in above describe
+   * way.
+   *
+   * @return complete url for telemetry service.
+   */
+  private String getTelemetryUrl() {
+    String telemetryBaseUrl = System.getenv(JsonKey.SUNBIRD_TELEMETRY_BASE_URL);
```
Comment (bvinayakumar) [inline actors/telemetry-core/src/main/java/org/sunbird/util/lmaxdisruptor/SunbirdTelemetryEventConsumer.java]: In project util, there is a method to get config value.

**Pull request merged**